examples/step-39/step-39.cc

```diff
@@ -859,16 +860,18 @@ namespace Step39
   }
 
 
-  // Some graphical output
+  // Create graphical output. We produce the filename by using a
+  // <code>std::ostringstream</code> object to assemble the name from
+  // its various components, including the refinement cycle that we
+  // output with two digits.
   template <int dim>
   void InteriorPenaltyProblem<dim>::output_results (const unsigned int cycle) const
   {
-    // Output of the solution in gnuplot format.
-    char *fn = new char[100];
-    sprintf(fn, "sol-%02d", cycle);
+    std::ostringstream filename_stream;
+    filename_stream << "sol-" << Utilities::int_to_string(cycle,2)
+                    << ".gnuplot";
+    const std::string filename = filename_stream.str();
```

Comment:
Since we are left with three strings at the end of the day we could make this a bit simpler by writing
```cpp
const std::string filename = "sol-" + Utilities::to_string(cycle, 2) + ".gnuplot";
```

instead of using the string stream.